### PR TITLE
code for 100% coverage of QuestionSetObjectFactory

### PR DESCRIPTION
**Pull request by alblin**
see issue #160 

code coverage result: 
screen shot 2017-04-17 at 12 55 27 pm

### CODE REVIEW

File: client/domain/QuestionSetObjectFactorySpec.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  describe('getQuestionsId', function() {
+    it('should get question id index', function() {
+      var dict = {questionIds: ["q0", "q1"]};
+      var questionSet = QuestionSetObjectFactory.create(dict);
+      var thisId = questionSet.getQuestionId(1);
+
+      expect(thisId).toMatch("q1");
+    });
+
+    it('should throw error if index out of bounds of number of questions', function() {
```
Comment (eyurko) [inline client/domain/QuestionSetObjectFactorySpec.js]: 'should throw error if index passed to getQuestionId is out of bounds'

File: client/domain/QuestionSetObjectFactorySpec.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    QuestionSetObjectFactory = $injector.get('QuestionSetObjectFactory');
+  }));
+
+  describe('getQuestionsIds', function() {
+    it('should get question ids', function() {
+      var dict = {questionIds: ["q0", "q1"]};
+      var questionSet = QuestionSetObjectFactory.create(dict);
+      var thisDict = questionSet.getQuestionIds();
```
Comment (eyurko) [inline client/domain/QuestionSetObjectFactorySpec.js]: var questionIds

File: client/domain/QuestionSetObjectFactorySpec.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  var QuestionSetObjectFactory;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    QuestionSetObjectFactory = $injector.get('QuestionSetObjectFactory');
+  }));
+
+  describe('getQuestionsIds', function() {
+    it('should get question ids', function() {
+      var dict = {questionIds: ["q0", "q1"]};
```
Comment (eyurko) [inline client/domain/QuestionSetObjectFactorySpec.js]: var questionSetDict

File: client/domain/QuestionSetObjectFactorySpec.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+      expect(thisDict).toEqual(dict.questionIds);
+    });
+  });
+
+  describe('getQuestionsId', function() {
+    it('should get question id index', function() {
+      var dict = {questionIds: ["q0", "q1"]};
+      var questionSet = QuestionSetObjectFactory.create(dict);
+      var thisId = questionSet.getQuestionId(1);
```
Comment (eyurko) [inline client/domain/QuestionSetObjectFactorySpec.js]: var questionId

File: client/domain/QuestionSetObjectFactorySpec.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      var questionSet = QuestionSetObjectFactory.create(dict);
+      var numberOfQuestions = questionSet.getNumberOfQuestions();
+
+      expect(numberOfQuestions).toEqual(2);
+    });
+  });
+
+  describe('getIntroductionParagraphs', function() {
+    it('should get introduction paragraph', function() {
+      var dict = {introductionParagraphs: "introduction"};
```
Comment (eyurko) [inline client/domain/QuestionSetObjectFactorySpec.js]: paragraphsDict

File: client/domain/QuestionSetObjectFactorySpec.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      var questionSet = QuestionSetObjectFactory.create(dict);
+      var introductionParagraphs = questionSet.getIntroductionParagraphs();
+
+      expect(introductionParagraphs).toMatch("introduction");
```
Comment (eyurko) [inline client/domain/QuestionSetObjectFactorySpec.js]: .toMatch(paragraphsDict['introductionParagraphs']).

**Comment by alblin** [comment]
@eyurko addressed your comments in most recent commit

**Pull request merged**